core/src/main/java/org/apache/iceberg/encryption/StandardEncryptionManager.java

```diff
@@ -92,13 +94,45 @@ public ByteBuffer wrapKey(ByteBuffer secretKey) {
 
   public ByteBuffer unwrapKey(ByteBuffer wrappedSecretKey) {
     if (kmsClient == null) {
-      throw new IllegalStateException(
-          "Cannot wrap key after called after serialization (missing KMS client)");
+      throw new IllegalStateException("Cannot unwrap key after serialization (missing KMS client)");
     }
 
     return kmsClient.unwrapKey(wrappedSecretKey, tableKeyId);
   }
 
+  public String wrappedKeyEncryptionKey() {
```

Comment:
As I noted above, I don't think that the encryption manager is the right place to track this key. I think that it should be part of `TableMetadata` or just part of `Snapshot`. The `EncryptionManager` should work with any version of the table and should not have mutable state based on the `TableMetadata` or `Snapshot` that it is currently working with.

Reply:
Per our discussion, we'll use a key cache for this. I'll update the code.